src/srf/boolean.cpp

```diff
@@ -20,20 +20,23 @@ void SShell::MakeFromIntersectionOf(SShell *a, SShell *b) {
     MakeFromBoolean(a, b, SSurface::CombineAs::INTERSECTION);
 }
 
-// We will be inserting existing verticies into curves to split them
-// todo: this is only using the ends of exact curves, and it is only
-// using them to split existing curves, not new intersections.
-// It resolves some issues but we could do better. We will need to
-// reorder things so the surface intersection curves exist prior to
-// splitting any curves at all in order to have their verticies too.
-// Better still would be to get curve/surface intersection to work
-// more reliably at the edges - maybe do curve/curve tests as part
-// of the curve-surface intersection test.
-static void FindVertsOnCurve(List<SInter> *l, const SCurve *curve, SShell *sh) {
-    for(auto sc : sh->curve) {
-        if(!sc.isExact) continue;
-        for(int i=0; i<2; i++) {
-            Vector pt = sc.exact.ctrl[ i==0 ? 0 : sc.exact.deg ];
+// We will be inserting existing surface verticies into curves to split them.
+// Using surface verts is faster than curve verts. Neither may be optimal but
+// doing this resolves some issues where curves are not split.
+static void FindVertsOnCurve(List<SInter> *l, const SCurve *curve, SShell *sh,
+                            Vector amin, Vector amax) {
+
+    for(auto srf : sh->surface) {
+        Vector bmax, bmin;
+        srf.GetAxisAlignedBounding(&bmax, &bmin);
+
+        if(Vector::BoundingBoxesDisjoint(amax, amin, bmax, bmin)) {
```

Comment:
Yes, this should bring the profiling to "before https://github.com/solvespace/solvespace/commit/ab10e38d44b26de3ee64b60db0d154278d1ee8ac state" where most of the time was spent on bounding boxes instead of on other vector:: methods.